### PR TITLE
update metrics

### PR DESCRIPTION
**Pull request by StevenLuMT**
Descriptions of the changes in this PR:

### Motivation

some metric's value is not right,so update it
the current is problem-driven, and a comprehensive review will be done later.

### Changes

update 2 metric:
1.Bookie: ReadBytes use entrySize
2.Journal: report journal write error metric

### CODE REVIEW

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieImpl.java [hunk 1]
```diff
@@ -1082,7 +1082,8 @@ public ByteBuf readEntry(long ledgerId, long entryId)
                 LOG.trace("Reading {}@{}", entryId, ledgerId);
             }
             ByteBuf entry = handle.readEntry(entryId);
-            bookieStats.getReadBytes().add(entry.readableBytes());
+            entrySize = entry.readableBytes();
+            bookieStats.getReadBytes().add(entrySize);
```
Comment (Shoothzj) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieImpl.java:1086]: unnecessary change?

Comment (StevenLuMT) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieImpl.java:1086]: it's necessary  for the finally using it

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieImpl.java [hunk 2]
```diff
@@ -1082,7 +1082,8 @@ public ByteBuf readEntry(long ledgerId, long entryId)
...
             success = true;
             return entry;
         } finally {
```
Comment (StevenLuMT) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieImpl.java:1089]: this code

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieImpl.java (hunk 1 above)
Comment (Shoothzj) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieImpl.java:1086]: I didn't see the `finally` cause refer the entrySize variable. Can you explain more detail? thanks

**Comment by StevenLuMT** [comment]
rerun failure checks

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieImpl.java (hunk 1 above)
Comment (StevenLuMT) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieImpl.java:1086]: > I didn't see the `finally` cause refer the entrySize variable. Can you explain more detail? thanks

I add the code picture

**Comment by StevenLuMT** [comment]
@eolivelli @pkumar-singh @zymap
If you have time, please help me review it, thank you

**Comment by StevenLuMT** [comment]
@eolivelli @pkumar-singh @zymap
If you have time, please help me review it, thank you

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerFragmentReplicator.java [hunk 3]
```diff
@@ -395,14 +395,15 @@ public void readComplete(int rc, LedgerHandle lh,
                 byte[] data = entry.getEntry();
                 final long dataLength = data.length;
                 numEntriesRead.inc();
-                numBytesRead.registerSuccessfulValue(dataLength);
 
                 ByteBufList toSend = lh.getDigestManager()
                         .computeDigestAndPackageForSending(entryId,
                                 lh.getLastAddConfirmed(), entry.getLength(),
                                 Unpooled.wrappedBuffer(data, 0, data.length));
+                int readBytes = toSend.readableBytes();
```
Comment (nicoloboschi) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerFragmentReplicator.java:403]: why did you modify this? 
`numBytesRead` tracks the data that have been read

**Review by nicoloboschi — CHANGES_REQUESTED** [review]

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerFragmentReplicator.java (hunk 3 above)
Comment (StevenLuMT) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerFragmentReplicator.java:403]: > why did you modify this?
> `numBytesRead` tracks the data that have been read

yes, this numBytesRead  is equal, I rollback this, thanks @nicoloboschi

**Comment by StevenLuMT** [comment]
rerun failure checks

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java [hunk 4]
```diff
@@ -397,6 +397,10 @@ public int process(boolean shouldForceWrite) throws IOException {
                 }
 
                 return forceWriteWaiters.size();
+            } catch (Throwable e) {
```
Comment (zymap) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java:400]:
```suggestion
            } catch (IOException e) {
```

**Comment by StevenLuMT** [comment]
rerun failure checks

**Review by dlg99 — APPROVED** [review]

**Review by zymap — APPROVED** [review]
